src/Connection.php

```diff
@@ -9,17 +9,39 @@ class Connection
 {
     use \atk4\core\DIContainerTrait;
 
+    const DEFAULT_DRIVER_TYPE = null;
+
+    /**
+     * @deprecated use $queryClass instead
+     */
+    protected $query_class;
+
     /** @var string Query classname */
-    protected $query_class = Query::class;
+    protected $queryClass = Query::class;
+
+    /**
+     * @deprecated use $expressionClass instead
+     */
+    protected $expression_class;
 
     /** @var string Expression classname */
-    protected $expression_class = Expression::class;
+    protected $expressionClass = Expression::class;
 
     /** @var Connection|\PDO Connection or PDO object */
+    protected $handler;
+
+    /**
+     * @deprecated use $handler instead
+     */
     protected $connection;
```

Comment:
This is protected, there does not need to be BC. We are not Google.

Reply:
The problem is `DIContainerTrait` lets setting protected properties from external locations.
Some of the legacy projects might use `['connection' => Whatever]` so I decided to keep it BC for time being.

Reply:
Wdyt to update `DIContainerTrait` which throw an exception (with hint) when deprecated property is set with it?

Or even a new core trait which will accept list of deprecated properties/methods and throw an exception when any of them are accessed (using magic methods/already existing trait).

Reply:
Yes, good idea!